tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -4826,38 +4826,56 @@ Status ConvertUnpack(OpConverterParams* params) {
   return ConvertSplitHelper(params, inputs.at(0), tf_axis, num, true);
 }
 
-// Supports cast fp16=>fp32 through IIdentityLayer.
+// Supports cast through IIdentityLayer.
 Status ConvertCast(OpConverterParams* params) {
   const NodeDef& node_def = params->node_def;
   TF_RETURN_IF_ERROR(CheckInputsWeights(*params, {{"x", false}}));
-  auto unsupport_cast_error = [&]() {
+  
+  auto unsupported_cast_error = [&](DataType in_dtype, DataType out_dtype) {
     return errors::Unimplemented("Cast op: ", node_def.op(),
-                                 " not supported at: ", node_def.name());
+                                 " not supported at: ", node_def.name(),
+                                 ". Input DType:", DataTypeString(in_dtype),
+                                 ". Output DType:", DataTypeString(out_dtype));
   };
 
   DataType input_type;
   TF_RETURN_IF_ERROR(GetInputTfType(*params, &input_type, 0));
-  if (input_type != DataType::DT_HALF) {
-    return unsupport_cast_error();
-  }
 
   DataType output_type;
   TF_RETURN_IF_ERROR(GetNodeDefTfType(params->node_def, &output_type,
                                       kCastOutputTypeAttrName));
 
-  if (output_type != DataType::DT_FLOAT) {
-    return unsupport_cast_error();
+  if (input_type != DataType::DT_FLOAT && input_type != DataType::DT_HALF) {
+      return unsupported_cast_error(input_type, output_type);
+  }
+  if (output_type != DataType::DT_FLOAT && output_type != DataType::DT_HALF) {
+      return unsupported_cast_error(input_type, output_type);
   }
 
+  #if !IS_TRT_VERSION_GE(7, 0, 0, 11)
```

Comment:
Shall we move this check before the other twos?

Reply:
Applied your recommendation